### PR TITLE
[filebeat][decode_cef] Add hyphen support

### LINKED ISSUE #40348: [filebeat][decode_cef] Unable to parse fields containing hyphen `-`
This is an extension to https://github.com/elastic/beats/issues/40236 where a workaround was performed before `decode_cef` processor as it is unable to handle fields containing hyphen `-`.

Sample message:

```python
2536 <14>1 2024-07-04T09:16:35.992Z logfwd20-251f92c6-abd9-4da9-a32f-ea60baed66ca-taskmanager-wx85p logforwarder - panwlogs - CEF:0|Palo Alto Networks|LF|2.0|TRAFFIC|end|3|dtz=UTC rt=Jul 04 2024 09:03:48 deviceExternalId=no-serial PanOSConfigVersion=10.2 start=Jul 04 2024 09:03:39 src=127.0.0.1 dst=0.0.0.0 sourceTranslatedAddress=0.0.0.0 destinationTranslatedAddress= cs1=intrazone-default cs1Label=Rule suser= duser= app=unknown-udp cs3=vsys1 cs3Label=VirtualLocation cs4=untrust cs4Label=FromZone cs5=untrust cs5Label=ToZone deviceInboundInterface=ethernet1/1 deviceOutboundInterface=ethernet1/1 cs6=Cortex Data Lake cs6Label=LogSetting cn1=574297 cn1Label=SessionID cnt=1 spt=13442 dpt=500 sourceTranslatedPort=0 destinationTranslatedPort=0 proto=udp act=allow PanOSBytes=82 out=82 in=0 cn2=1 cn2Label=PacketsTotal PanOSSessionStartTime=Jul 04 2024 09:03:05 cn3=0 cn3Label=SessionDuration cs2=any cs2Label=URLCategory externalId=7361339208201408573 PanOSSourceLocation=DE PanOSDestinationLocation=US PanOSPacketsSent=1 PanOSPacketsReceived=0 reason=aged-out PanOSDGHierarchyLevel1=65544 PanOSDGHierarchyLevel2=65545 PanOSDGHierarchyLevel3=65546 PanOSDGHierarchyLevel4=65550 PanOSVirtualSystemName= dvchost=GP cloud service cat=from-policy PanOSSourceUUID= PanOSDestinationUUID= PanOSIMSI=0 PanOSIMEI= PanOSParentSessionID=0 PanOSParentStarttime=Jan 01 1970 00:00:00 PanOSTunnel=N/A PanOSEndpointAssociationID=0 PanOSChunksTotal=0 PanOSChunksSent=0 PanOSChunksReceived=0 PanOSRuleUUID=40b8842f-eec3-4e04-b760-6a2ce4698fde PanOSHTTP2Connection=0 PanOSLinkChangeCount=0 PanOSSDWANPolicyName= PanOSLinkSwitches= PanOSSDWANCluster= PanOSSDWANDeviceType= PanOSSDWANClusterType= PanOSSDWANSite= PanOSDynamicUserGroupName= PanOSX-Forwarded-ForIP= PanOSSourceDeviceCategory= PanOSSourceDeviceProfile= PanOSSourceDeviceModel= PanOSSourceDeviceVendor= PanOSSourceDeviceOSFamily= PanOSSourceDeviceOSVersion= PanOSSourceDeviceHost= PanOSSourceDeviceMac= PanOSDestinationDeviceCategory= PanOSDestinationDeviceProfile= PanOSDestinationDeviceModel= PanOSDestinationDeviceVendor= PanOSDestinationDeviceOSFamily= PanOSDestinationDeviceOSVersion= PanOSDestinationDeviceHost= PanOSDestinationDeviceMac= PanOSContainerID= PanOSContainerNameSpace= PanOSContainerName= PanOSSourceEDL= PanOSDestinationEDL= PanOSGPHostID= PanOSEndpointSerialNumber= PanOSSourceDynamicAddressGroup= PanOSDestinationDynamicAddressGroup= PanOSHASessionOwner= PanOSTimeGeneratedHighResolution=Jul 04 2024 09:03:39 PanOSNSSAINetworkSliceType= PanOSNSSAINetworkSliceDifferentiator=
```

If `decode_cef` is applied to above message, we get error: `malformed value for PanOSDynamicUserGroupName at pos 1617`, because it is unable to parse adjacent field `PanOSX-Forwarded-ForIP`. When a workaround is applied to remove hyphen `-` from the field name, this error is resolved. 
Below is the filebeat configuration with current workaround (removing hyphen `-` from fields) to mitigate the errors.

Filebeat configuration:
```python
filebeat.inputs:
- type: tcp
  host: "127.0.0.1:9528"
  processors:
  - copy_fields:
      fields:
        - from: message
          to: event.original
      fail_on_error: true
      ignore_missing: false
  - replace:
      fields:
        - field: "message"
          pattern: "PanOSX-Forwarded-ForIP="
          replacement: "PanOSXForwardedForIP="
        - field: "message"
          pattern: "PanOSX-Forwarded-For="
          replacement: "PanOSXForwardedFor="
        - field: "message"
          pattern: "PanOSSplit-tunnelconfiguration="
          replacement: "PanOSSplitTunnelconfiguration="
      ignore_missing: false
      fail_on_error: true
  - decode_cef:
      field: message
      ecs: false
  - drop_fields:
      fields: ["message"]
      ignore_missing: false
```

### PR DESCRIPTION
**Pull request by vinit-chauhan**
## Proposed commit message

This PR Adds support for `-` in the extension keys.
To do this, I have updated the `cef.rl` file and used `go generate` to generate `parser.go` and `parser_recover.go`.
Moreover, I have also added .ri file to gitignore file as they are intermediate files generated by regel. 

~~**Note: I've also updated the go generate directive to use regel-go as the latest version is not supporting `-Z` flag**~~

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #40348 

## Screenshots

Tests Passing
image

### CODE REVIEW

File: x-pack/filebeat/processors/decode_cef/cef/cef.go [hunk 1]
```diff
@@ -13,9 +13,9 @@ import (
 )
 
 // Parser is generated from a ragel state machine using the following command:
-//go:generate ragel -Z -G1 parser.rl -o parser.go
+//go:generate ragel-go -G1 parser.rl -o parser.go
```
Comment (andrewkroh) [inline x-pack/filebeat/processors/decode_cef/cef/cef.go:16]: The diff is much larger than I would have expected for this change. What ragel version did you use?

6.10 from http://www.colm.net/open-source/ragel/ (which is also what [brew gives](https://formulae.brew.sh/formula/ragel))?

Comment (vinit-chauhan) [reply x-pack/filebeat/processors/decode_cef/cef/cef.go:16]: I'm using the latest version. 7.0.4
image

Comment (vinit-chauhan) [reply x-pack/filebeat/processors/decode_cef/cef/cef.go:16]: FWIW, parser.go and parser_recover.go are generated files. Those two have major changes among all the files.  The only manual change to those two is the comment stating Elastic License and fixed vetting error.

Comment (andrewkroh) [reply x-pack/filebeat/processors/decode_cef/cef/cef.go:16]: If the website is correct then 6.10 is the stable version and 7.0.4 is a development version (unstable). We should use the stable version.

Comment (vinit-chauhan) [reply x-pack/filebeat/processors/decode_cef/cef/cef.go:16]: Sure, let me downgrade it and regenerate the files

Comment (vinit-chauhan) [reply x-pack/filebeat/processors/decode_cef/cef/cef.go]: Hey @andrewkroh - seems like v6.10 was the key. Now the generated files have only a few changes. 
And `-Z` option is now available in ragel;  `ragel-go` is a v7 thing.

Comment (vinit-chauhan) [reply x-pack/filebeat/processors/decode_cef/cef/cef.go]: I've updated the code. Additionally, should we mention somewhere that we are allowing extra chars on top of CEF Naming guidelines?

File: CHANGELOG-developer.next.asciidoc [hunk 2]
```diff
@@ -199,6 +199,7 @@ The list below covers the major changes between 7.0.0-rc2 and main only.
 - Added `ignore_empty_values` flag in `decode_cef` Filebeat processor. {pull}40268[40268]
 - Bump version of elastic/toutoumomoma to remove internal forks of stdlib debug packages. {pull}40325[40325]
 - Refactor x-pack/filebeat/input/websocket for generalisation. {pull}40308[40308]
+- Added support for hyphens in extension keys in `decode_cef` Filebeat processor. {pull}40427[40427]
```
Comment (andrewkroh) [inline CHANGELOG-developer.next.asciidoc:202]: This one isn't needed. Can you revert it please. The intent behind this file is to signal changes about the project tooling or important API changes. The addition to CHANGELOG.next.asciidoc is all this change needs.

**Review by andrewkroh — COMMENTED** [review]
LGTM.

**Comment by vinit-chauhan** [comment]
Hey @andrewkroh - are we waiting for code owners' review?

**Comment by andrewkroh** [comment]
Yeah, I was giving the little time in case anyone from @elastic/sec-deployment-and-devices wanted to comment. I'll merge it after CI finishes today.

**Comment by andrewkroh** [comment]
run docs-build

**Review by pkoutsovasilis — APPROVED** [review]
Thank you for this contribution @vinit-chauhan! LGTM

**Comment by pkoutsovasilis** [comment]
run docs-build

**Pull request merged**